### ISSUE TITLE
building the git version in MSYS2

### ISSUE THREAD

**Issue by trolzen** [premake/premake-core#1869]
How do I build the git version of Premake using `Bootstrap.mak` in MSYS2? The closest target platform is `mingw` but the following command fails:
```
user@host MINGW64 /c/projects/premake5/premake-5.0.0-beta1-src
$ make -f Bootstrap.mak  mingw
if exist .\bin rmdir /s /q .\bin
/bin/sh: -c: line 2: syntax error: unexpected end of file
make: *** [Bootstrap.mak:80: windows-clean] Error 2
```

Looks like it fails because of the windows-style commands in the `windows-clean` target. Shouldn't the `mingw-clean` inherit from the `nix-clean` instead of current `windows-clean`?
https://github.com/premake/premake-core/blob/0c6711170c60c208e15573a749b4fd21d5b28f6a/Bootstrap.mak#L84

I know that Premake is in the MSYS2 packages and that I can install it from there, but still.

**Comment by simbit18** [comment]
You can try to add in Bootstrap.mak this:

add in .PHONY: => msys2-clean msys2

.PHONY: default none clean nix-clean windows-clean \
	mingw-clean mingw msys2-clean msys2 macosx macosx-clean osx-clean osx \
	linux-clean linux bsd-clean bsd solaris-clean solaris \ 
	haiku-clean haiku windows-base windows windows-msbuild

msys2-clean: nix-clean

msys2: msys2-clean
	mkdir -p build/bootstrap
	$(CC) -o build/bootstrap/premake_bootstrap -DPREMAKE_NO_BUILTIN_SCRIPTS -I"$(LUA_DIR)" -I"$(LUASHIM_DIR)" $(SRC) -lole32 -lversion
	./build/bootstrap/premake_bootstrap embed
	./build/bootstrap/premake_bootstrap --arch=$(PLATFORM) --os=windows --to=build/bootstrap --cc=mingw gmake2
	$(MAKE) -C build/bootstrap config=$(CONFIG)_$(PLATFORM:x86=win32)

**Comment by Biswa96** [comment]
Hence msys2's mingw-w64-premake has this patch to workaround this issue https://github.com/msys2/MINGW-packages/blob/bf3b2b0e5d8b49e60ea18b349c84dba6dca2a190/mingw-w64-premake/0003-bootstrap-use-nix-commands.patch

**Comment by samsinsane** [comment]
I'm not that familiar with MSYS2, MinGW or Cygwin, mostly I'm just aware they exist. There was a PR that changed the MinGW cleanup from `rm` to `rmdir` - do any of these support `rmdir`? Do they all support `rm`? Is there a situation where changing to `rm` would break something for someone?

I'm happy to see it change if it's causing unnecessary friction, but I'd rather we avoided having a chain of PRs flipping between `rm` and `rmdir`.

Semi-related, I tried to get MinGW builds going for Premake at the start of the year and I ended up needing to make this change, so if this is the correct change I can probably open a PR and we can have MinGW64 builds running. The commit if anyone has any pre-PR feedback: https://github.com/LORgames/premake-core/commit/1d58ef4814dc8da5f7aaf1cb61e89c55c55a5654
